### PR TITLE
Transaction feed details

### LINKED ISSUE #1572: Transaction Feed: Add multi-token support to Transaction Details Screen
In #1262 we started implementing a new Transaction Feed with multi-token support.

When tapping on an item, it's not open the detail screen. We need to implement it and open it. We can either reuse existing ones or create a new one.

Work on this is in progress in this PR: https://github.com/valora-inc/wallet/tree/transaction-feed-details
For now it works for the Payment Sent transactions but is missing payments received, invite sent/received and exchanging. Also tests.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Add transaction detail screens for multi-token support.

### Other changes

N/A

### Tested

Manually and with unit tests

### How others should test

Open different types of txs (incoming transfers, outgoing transfers, invites, rewards receptions) and check that everything looks as it used to/expected

### Related issues

- Fixes #1572

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  const tokensByCurrency = useSelector(tokensByCurrencySelector)
+  const celoAddress = tokensByCurrency[Currency.Celo]?.address
+  const rewardsSenders = useSelector(rewardsSendersSelector)
+  const addressToDisplayName = useSelector(addressToDisplayNameSelector)
+
+  switch (transaction.type) {
+    case TokenTransactionTypeV2.Exchange:
+      // TODO: Change this to show any exchanges, not just CELO sell/purchase.
+      const isCeloSell = (transaction as TokenExchange).inAmount.tokenAddress === celoAddress
+      return isCeloSell ? i18n.t('soldGold') : i18n.t('purchasedGold')
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx:39]:
```suggestion
      return isCeloSell ? t('soldGold') : t('purchasedGold')
```

File: packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    case TokenTransactionTypeV2.Sent:
+      const isCeloSend = (transaction as TokenTransfer).amount.tokenAddress === celoAddress
+      return t(isCeloSend ? 'transactionHeaderWithdrewCelo' : 'transactionHeaderSent')
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx:42]: We should prefer static `t()` expressions to more easily migrate to a tool extracting strings from the codebase.
```suggestion
      return isCeloSend ? t('transactionHeaderWithdrewCelo') : t('transactionHeaderSent')
```

File: packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      const isCeloReception = transfer.amount.tokenAddress === celoAddress
+      if (
+        rewardsSenders.includes(transfer.address) ||
+        addressToDisplayName[transfer.address]?.isCeloRewardSender
+      ) {
+        return t('transactionHeaderCeloReward')
+      } else {
+        return isCeloReception
+          ? i18n.t('transactionHeaderCeloDeposit')
+          : i18n.t('transactionHeaderReceived')
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx:54]:
```suggestion
          ? t('transactionHeaderCeloDeposit')
          : t('transactionHeaderReceived')
```

File: packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+  const addressToDisplayName = useSelector(addressToDisplayNameSelector)
+  const rewardsSenders = useSelector(rewardsSendersSelector)
+
+  let content
+
+  switch (transaction.type) {
+    case TokenTransactionTypeV2.Exchange:
+      // TODO:
+      content = null
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx:82]: Will be in a follow up PR?

File: packages/mobile/test/values.ts [hunk 5]
```diff
@@ -56,7 +56,7 @@ export const mockE164NumberHashWithPepper =
   '0xf6429456331dedf8bd32b5e3a578e5bc589a28d012724dcd3e0a4b1be67bb454'
 
 export const mockE164Number2 = '+12095559790'
-export const mockDisplayNumber2 = '(209) 555-9790'
+export const mockDisplayNumber2 = '+1 209-555-9790'
```
Comment (jeanregisser) [inline packages/mobile/test/values.ts:59]: Why is this change needed?

File: packages/mobile/src/transactions/feed/detailContent/TransferSentContent.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      <UserSection
+        type={isCeloWithdrawal ? 'withdrawn' : 'sent'}
+        recipient={recipient}
+        avatar={<TransferAvatars type="sent" recipient={recipient} />}
+        testID="TransferSent"
+      />
+      <CommentSection comment={metadata.comment} />
+      <HorizontalLine />
+      <LineItemRow
+        title={t(isCeloWithdrawal ? 'amountCeloWithdrawn' : 'amountSent')}
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/feed/detailContent/TransferSentContent.tsx:54]:
```suggestion
        title={isCeloWithdrawal ? t('amountCeloWithdrawn') : t('amountSent')}
```

**Review by jeanregisser — COMMENTED** [review]
Looking great! 👍 
Left some small nits/questions.

File: packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx (hunk 4 above)
Comment (gnardini) [reply packages/mobile/src/transactions/feed/TransactionDetailsScreen.tsx:82]: Yes, the issue is here: https://app.zenhub.com/workspaces/retention-squad-sprint-board-612f2ef98ef2b800116e6043/issues/valora-inc/wallet/1595

File: packages/mobile/test/values.ts (hunk 5 above)
Comment (gnardini) [reply packages/mobile/test/values.ts:59]: The existing mock was not used anywhere and when displaying it showed up like this. I just thought it was an oversight, is this not how it's supposed to be formatted? Argentinian numbers have always showed up following this pattern, but maybe american ones are different, although I didn't change anything related to the number formatting.

Comment (jeanregisser) [reply packages/mobile/test/values.ts:59]: Thanks all good then.

**Review by jeanregisser — APPROVED** [review]

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this task using Latest Android Internal build and Test flight build V1.24.1 and added the observation in below sheet.

**Sheet Link**:
https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=2039480032

**Devices** : Galaxy M40 (10.0), Google Pixel 2XL (11.0), iPhone 12 (14.7.1), iPhone 13 mini (15.1.1)

CC: @gnardini
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this task using Latest Android Internal release build V1.24.1 and added the observation in below sheet.

**Sheet Link**:
https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=2039480032

**Devices** : Samsung Galaxy A8+(8.0)

CC: @gnardini
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this task using Latest Test flight release build V1.24.1 and added the observation in below sheet.

**Sheet Link**:
https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=2039480032

**Devices** : iPhone 7(13.3)

CC: @gnardini
Thanks..!!
